salt/etcd/update-pre-reboot.sls

```diff
@@ -0,0 +1,31 @@
+{% set roles = salt['grains.get']('roles', []) %}
+{% set has_etcd_role = ("etcd" in roles) %}
+
+{% set is_etcd_member = salt['file.directory_exists' ]('/var/lib/etcd/member') and
+                    not salt['file.directory_exists' ]('/var/lib/etcd/proxy') %}
+
+{%- if is_etcd_member and not has_etcd_role -%}
+  # this is really running a member of the etcd cluster but it doesn't
+  # have the 'etcd' role: set the 'etcd' role so we are sure it will be
+  # running etcd after the update
+
+add-etcd-role:
+  grains.append:
+    - name: roles
+    - value: etcd
+
+{% else %}
```

Comment:
this else is trigegring, and causing the issue - http://pastebin.nue.suse.com/18211/src